contracts/contracts/ERC677/starkgate_token.cairo

```diff
@@ -0,0 +1,195 @@
+%lang starknet
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+from starkware.cairo.common.alloc import alloc
+from starkware.cairo.common.math import assert_not_zero, assert_le
+from starkware.cairo.common.uint256 import (
+    Uint256,
+    uint256_add,
+    uint256_check,
+    uint256_le,
+    uint256_lt,
+    uint256_sub,
+)
+from starkware.starknet.common.syscalls import get_caller_address
+from starkware.cairo.common.bool import FALSE, TRUE
+from openzeppelin.introspection.erc165.library import ERC165
+from contracts.starkware.starknet.std_contracts.ERC20.ERC20_base import (
+    ERC20_allowances,
+    ERC20_approve,
+    ERC20_burn,
+    ERC20_initializer,
+    ERC20_mint,
+    ERC20_transfer,
+    allowance,
+    balanceOf,
+    decimals,
+    name,
+    symbol,
+    totalSupply,
+)
+from contracts.starkware.starknet.std_contracts.ERC20.permitted import (
+    permitted_initializer,
+    permitted_minter,
+    permitted_minter_only,
+    permittedMinter,
+)
+from contracts.starkware.starknet.std_contracts.upgradability_proxy.initializable import (
+    initialized,
+    set_initialized,
+)
```

Comment:
What's the point of this initialization here?

Below in the constructor we `set_initialized()`, why? Looks like we never check it again.

Reply:
All of this is just a direct copy of https://github.com/starkware-libs/starkgate-contracts/blob/main/src/starkware/starknet/std_contracts/ERC20/ERC20.cairo